backend/applications.go

```diff
@@ -0,0 +1,257 @@
+package main
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"net/http"
+	"strings"
+
+	"github.com/elonsoc/center/backend/service"
+	"github.com/go-chi/chi/v5"
+	_ "github.com/lib/pq"
+)
+
+/*
+	This file deals with application registration and everything that comes with it.
+	- recieving the registration form
+	- generating a project ID
+	- generating an API key
+	- storing the information in the database
+	- returning the pertinent information to the user
+
+	Database Design can be found: https://docs.google.com/document/d/1zEK9K7crTcCcE9bMKm87qRHCyqa5I0_vjuI9eqYSaLg/edit?usp=sharing
+*/
+
+// Defining the struct for the applications router
+type ApplicationsRouter struct {
+	chi.Router
+	Svcs *service.Service
+}
+
+// NewApplicationsRouter is a function that returns a new applications router
+func NewApplicationsRouter(a *ApplicationsRouter) *ApplicationsRouter {
+	a.Svcs.Logger.Info("Initializing applications router")
+
+	r := chi.NewRouter()
+	r.Post("/", a.newApp)
+	r.Get("/", a.myApps)
+
+	a.Router = r
+	a.Svcs.Logger.Info("Applications router initialized")
+	return a
+}
+
+// create a registration structure (This is not entirely necessary, but it makes things easier)
+// Contents of this might change as the project progresses
+type application struct {
```

Comment:
You might want to make this an exported struct _and_ export each attribute of this struct so it can be used outside of the package. Also, you should add [struct tags](https://www.digitalocean.com/community/tutorials/how-to-use-struct-tags-in-go) to each attribute that match the [json](https://www.digitalocean.com/community/tutorials/how-to-use-struct-tags-in-go#encoding-json) and [database](https://upper.io/v4/getting-started/struct-mapping/) names.
```suggestion
type application struct {
    AppName string `json:"name" db:"name"`
    .
    .
    IsValid string `json:"is_valid,omitempty" db:"isValid"`
```

Reply:
Okay so I have done this. Currently all three (the go struct, the json struct tag, and the db struct tag) are different cases. Go is PascalCase so that it can exported like you said, json is camelCase, and db is snake_case. I came to this conclusion after looking into what is to be considered as "industry standard" for each "language". Could you explain to me the benefits of switching isValid to a string and giving it the omitempty struct tag, instead of just leaving it as a bool?

Reply:
Ah, you don't have to add `omitempty`, I believe I was just showing that you could use multiple tags.

Reply:
I see. In any case, this is resolved.